addons/project_todo/models/res_users.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+
+from odoo import _, api, fields, models, modules
+
+
+class Users(models.Model):
+    _name = 'res.users'
+    _inherit = ['res.users']
+
+    @api.model
+    def systray_get_activities(self):
+        """ Split To-do and Project activities in systray by removing
+            the single project.task activity represented and doing a
+            new query to split them between private/non-private tasks.
+        """
+        activity_groups = super().systray_get_activities()
+        # 1. removing project.task activity group
+        for group in activity_groups:
+            if group.get('model') == 'project.task':
+                activity_groups.remove(group)
+                break
+
+        # 2. creating groups for todo and task seperately
+        query = """SELECT BOOL(t.project_id) as is_task, count(*), act.res_model as model, act.res_id,
+                       CASE
+                           WHEN CURRENT_DATE - act.date_deadline::date = 0 Then 'today'
+                           WHEN CURRENT_DATE - act.date_deadline::date > 0 Then 'overdue'
+                           WHEN CURRENT_DATE - act.date_deadline::date < 0 Then 'planned'
+                        END AS states
+                     FROM mail_activity AS act
+                     JOIN project_task AS t ON act.res_id = t.id
+                    WHERE act.res_model = 'project.task' AND act.user_id = %(user_id)s
+                 GROUP BY is_task, states, act.res_model, act.res_id;
+                """
+        self.env.cr.execute(query, {
+            'user_id': self.env.uid,
+        })
+        activity_data = self.env.cr.dictfetchall()
+
+        user_activities = {}
+        for activity in activity_data:
+            is_task = activity['is_task']
+            if is_task not in user_activities:
+                if not is_task:
+                    module = 'project_todo'
+                    name = _('To-Do')
+                else:
+                    module = 'project'
+                    name = _('Task')
+                icon = modules.module.get_module_icon(module)
+                user_activities[is_task] = {
+                    'id': self.env['ir.model']._get('project.task').id,
+                    'name': name,
+                    'model': 'project.task',
+                    'type': 'activity',
+                    'icon': icon,
+                    'total_count': 0, 'today_count': 0, 'overdue_count': 0, 'planned_count': 0,
+                    'res_ids': set(),
+                }
+            user_activities[is_task]['res_ids'].add(activity['res_id'])
+            user_activities[is_task][f"{activity['states']}_count"] += activity['count']
+            if activity['states'] in ('today', 'overdue'):
+                user_activities[is_task]['total_count'] += activity['count']
+
+        for group_type in user_activities:
```

Comment:
```suggestion
        for user_activity in user_activities.values():
```
We use the key only to access the value, so we might as well iterate on the values, no?